base/gmp.jl

```diff
@@ -69,6 +69,96 @@ function __init__()
     end
 end
 
+
+module MPZ
+# wrapping of libgmp functions
+# "output parameters" are labeled x, y, z, and are returned when appropriate
+# constant input parameter are labeled a, b, c
+
+using Base.GMP: BigInt, Limb
+
+gmpz(op::Symbol) = (Symbol(:__gmpz_, op), :libgmp)
+
+sizeinbase(a::BigInt, b) = Int(ccall((:__gmpz_sizeinbase, :libgmp), Csize_t, (Ptr{BigInt}, Cint), &a, b))
+
+for op in (:add, :sub, :mul, :fdiv_q, :tdiv_q, :fdiv_r, :tdiv_r, :gcd, :lcm, :and, :ior, :xor)
+    @eval begin
+        $op(x::BigInt, a::BigInt, b::BigInt) = (ccall($(gmpz(op)), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b); x)
```

Comment:
Yes you are right, though it is mitigated by the fact that `Base.gcd` is not touched here, we introduce only a hard-to-find `Base.GMP.MPZ.gcd`.